src/openms/include/OpenMS/ANALYSIS/ID/IDScoreSwitcherAlgorithm.h

```diff
@@ -98,6 +145,9 @@ namespace OpenMS
         {
           // TODO: find a better way to check if old score type is something different (even if it has same name)
           // This currently, is a workaround for e.g., having Percolator_qvalue as meta value and same q-value as main score (getScore()).
+          //Note by jpfeuffer: The problem with this is, that this may add the old score to some of the hits if different, but not
+          // all, in case one is by chance the same. I would be fine with this, if it was done in the beginning and checked
```

Comment:
can you add some details here what to do in the beginning and what to check? Might increase chances to have it picked up and added by someone.

Reply:
This existed already. It is just obvious that it has the aforementioned flaw.

You will not be able to check it in this low-level function. You must find all usages and check that beforehand.

Reply:
Too many options to implement this in a cleaner way.

Reply:
ok if it is not possible here then it would be practical to state it clearly as “contract” in the documentation

Reply:
Yes. But unfortunately I am pretty sure the contract is not honored currently 😂